### PR TITLE
Add ability to build as Houdini plugin

### PR DESCRIPTION
**Pull request by hshakula**
Inside you will find:
- Established cmake schema to build plugin either as usdview plugin or Houdini plugin by setting `PXR_BUILD_AS_HOUDINI_PLUGIN` with an appropriate value
- Basic implementation for Hydra AOV system, but disabled yet
- Ability to enable RPR tracing using environment variables

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,30 @@ set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/defaults
                       ${CMAKE_SOURCE_DIR}/cmake/macros)
 
 include(Options)
+
+if(PXR_BUILD_AS_HOUDINI_PLUGIN)
```
Comment (bsavery) [inline CMakeLists.txt]: We should probably namespace this as RPR_BUILD_AS_HOUDINI_PLUGIN.  PXR-* is for pixar/USD

File: README.md [hunk 2]
```diff
@@ -1,86 +1,95 @@
-AMD Radeon ProRender delegate for USD (hdRpr)
+AMD Radeon ProRender rendering plugin for USD Hydra
```
Comment (bsavery) [inline README.md]: Technically this is a "render delegate".  Maybe that would be a good way instead of plugin.

File: CMakeLists.txt (hunk 1 above)
Comment (hshakula) [reply CMakeLists.txt]: I added `PXR_BUILD_AS_HOUDINI_PLUGIN` guided by `PXR_ENABLE_RIF_SUPPORT`. Perhaps, in that case, we should consider renaming this variable as well?

File: README.md (hunk 2 above)
Comment (hshakula) [reply README.md]: It's actually a delegate, but technically it's a plugin. So maybe let's name it as USD team [names](https://graphics.pixar.com/usd/docs/api/hd_embree_page_front.html) embree delegate? What do you think?
> HdEmbree : Embree-based hydra renderer plugin.

File: CMakeLists.txt (hunk 1 above)
Comment (bsavery) [reply CMakeLists.txt]: Yes, that should be renamed as well...

**Comment by hshakula** [comment]
I'm closing this PR due to its size and then going to create smaller PRs

**Pull request closed without merging**